UT-Georeferencing-Alignment-Tin-12d-1

LeeGregory12d opened the pull request:
## Unit test

The UT reads in an IFC 4x1 file, with and without using its IfcMapConversion, and then writes out the data in IFC 4x3 in
(a)  Map coordinates with a unity IfcMapConversion.
(b) Truncated Coordinates with an IfcMapConversion to make them map coordinates.  

This is to demonstrate the three fundamental concepts needed for Civil BIM: georeferencing, alignments and tins.

## Covered topics

*tick one/many*

- [x] project setup
- [x] georeferencing
- [x] alignment semantics
- [ ] alignment geometry
- [ ] linear placement
- [ ] project breakdown structure / spatial structure
- [ ] terrain
- [x] triangulated irregular network
- [ ] open cross profile
- [ ] sectioned surface & solid
- [ ] pavement & course
- [ ] surface feature
- [ ] earthworks
- [ ] geotechnics
- [ ] longitudinal products (barriers, guardrail, ...)
- [ ] interchange
- [ ] structural
- [ ] drainage
- [ ] signage
- [ ] road furniture
- [ ] rail furniture
- [ ] ports & waterways furniture
- [ ] bridge furniture

LeeGregory12d committed:
UT-Georeferencing-Alignment-Tin-12d-1

LeeGregory12d committed:
Update readme

LeeGregory12d committed:
Updating readme with image

pjanck commented on UT-Georeferencing-Alignment-Tin-12d-1/readme.md:
```diff
@@ -0,0 +1,69 @@
+and the file has an IfcMapConversion to translate the data from Truncated Coordinates to Map Coordinates (Eastings and Northings).
+There was no 2D rotation or change of scale for X and Y in the IfcMapConversion.
+
+I don't know if the coordinates in the TUM file were truncated map coordinates or local engineerng coordinates.
+But being a real Civil project I suspect they are truncated map coordindates.
```
You are correct. Also, some typos.

```suggestion
I don't know if the coordinates in the TUM file were truncated map coordinates or local engineering coordinates.
But being a real Civil project I suspect they are truncated map coordinates.
```

LeeGregory12d replied:
fixed

pjanck commented on UT-Georeferencing-Alignment-Tin-12d-1/readme.md:
```diff
@@ -0,0 +1,69 @@
+|----------------------------------------------------------------|---------------------------------------------------------------------------------------|
+| `terrain-surface-ifcAlignment1x1-fixed-names.ifc`              | TUM IFC 4x1 file to import                                                            |
+| `ex-terrain-alignment.png`                                     | image of TUM file in TUM software                                                     |
+| `UT-Georeferencing-Alignment-Tin-12d-1.png`                    | image of TUM data in 12d Model (a) with Truncated Coordinates (b) and Map Coordinates |
+|                                                                |     and long sections of each aligment with the two tins                              |
```
Line breaks are handled by markdown itself, no need to do them.

LeeGregory12d replied:
fixed

LeeGregory12d replied:
I think it  is all now OK

pjanck commented on UT-Georeferencing-Alignment-Tin-12d-1/readme.md:
```diff
@@ -0,0 +1,69 @@
+For the alignments there is only the IFC semantic definition and no IFC geometry.
+The IFC 4x3 files were generated by 12d Model. 
+
+![](./UT-Georeferencing-Alignment-Tin-12d-1.png)
```
Suggestion: including the originating screen shot would be beneficial as well.

pjanck reviewed (changes requested):
Small readme improvements.

LeeGregory12d committed:
Update readme - learning how to include images

pjanck commented:
Following the changes to the schema, (we are now at `RC3`), we ask you to update your IFC files accordingly. The schema can be found [here](https://github.com/bSI-InfraRoom/IFC-Specification/blob/main/IFC4x3_RC3.exp). The checking procedure for IFC files has been updated - please merge in the current state of the `main` branch.

LeeGregory12d committed:
Update Georeferencing-Alignment-Tin-12d-1 to RC3

Update IFC files to RC3
Remove "UT-" from file names

LeeGregory12d committed:
Updated readme for Georeferencing-Alignment-Tin-12d-1

LeeGregory12d committed:
Fixed vertical segment errors in Georeferencing-Alignment-Tin-12d-1

Start chainage wasn't being substracted from the IfcVerticatSegments
Changed name of a file
